### ISSUE TITLE
Define escalation practices when there are hub outages

### ISSUE THREAD

**Issue by choldgraf** [2i2c-org/infrastructure#1118]
### Background and proposal

Hubs will experience outages of different magnitudes, and these should trigger varying degrees of response from our team. We want to find a balance between sustainable practices for our team, and ensuring that our communities don't feel too much pain from outages.

We should define an escalation protocol and pathway that we all agree upon. This could be a combination of support practices + notification practices, as well as setting expectations for ourselves and others about what to expect.

### Implementation guide and constraints

This issue is related to our first/second-tier support processes, and the issue below has some useful ideas about defining "seriousness" of support requests:

- https://github.com/2i2c-org/infrastructure/issues/1068

This issue is more focused on defining what we do when there are active problems, and what team processes we follow to solve them.

**Comment by sgibson91** [comment]
I think the discussion in Slack @yuvipanda and I had around https://www.pagerduty.com/ and https://www.atlassian.com/software/opsgenie will be relevant here. The idea being that for major outages, there should be an engineer on-call to respond so that the support steward isn't the only one who shoulders responsibility for updating the client and exploring the issue.

**Comment by damianavila** [comment]
> support steward isn't the only one who shoulders responsibility for updating the client and exploring the issue.

IMHO, the support steward should not "feel" responsible for exploring the issue (although it might be involved in updating the client), I guess this distinction is actually part of the discussion in #1068.

**Comment by sgibson91** [comment]
@damianavila sure, but I've certainly been posting in the Slack channel about outages and just had to do my best until someone came online. I think having someone to page will help that feeling of "I can't do anything more right now".

**Comment by choldgraf** [comment]
Given that we have a PR open to define and incident commander and more complex response process:

- https://github.com/2i2c-org/team-compass/pull/422

Should we re-scope this issue to explicitly be about "pager-style" escalation practices? E.g., some system to ping a specific person via a non-Slack method if a particular problem emerges?

Or, should we consider https://github.com/2i2c-org/team-compass/pull/422 to be enough, and we should close this and iterate with that system for a bit and decide if we need something like a dedicated Pager?

**Comment by yuvipanda** [comment]
> Should we re-scope this issue to explicitly be about "pager-style" escalation practices?

Yes! I think this is important as otherwise the 'currently awake' people can feel pretty overwhelmed sometimes. 

I think specifically for outages, as a short-term non-scaleable measure, I'm always happy to be alerted via non-slack methods (I think most people have my phone number). I think that's an important senior engineer responsibility.

**Comment by damianavila** [comment]
> I think that's an important senior engineer responsibility.

I totally appreciate that @yuvipanda, BUT we need to find a way/process so we do not need to ping you at your personal phone number. So +1 on repurposing this one about the "pager-style" tool.

**Comment by choldgraf** [comment]
OK I've re-worked the top comment in this one to focus more around Pager-style updates. Also added some links!

**Comment by yuvipanda** [comment]
@damianavila totally agree this isn't long term sustainable! I just wanted to volunteer that right now as outages and escalations will continue to happen as we figure out the process.

**Comment by yuvipanda** [comment]
I'm also looking at OpsGenie - In particular, they have a 'quiet hours' features that seems more intuitive than Pagerduty's scheduling. However, the OpsGenie slack integration doesn't let you trigger a new incident from slack which is boo :(

**Comment by yuvipanda** [comment]
It's absolutely important that engineers can control how and *when* they are notified - we don't want this to become a traditional 'oncall' situation.

**Comment by yuvipanda** [comment]
Me and @jmunroe spent a bunch of time talking about this, and in particular role-playing how this *should* have played out with today's UToronto outage.

A proposed workflow is that after the incident is created, we want pagerduty to evaluate the current local timezone and stated preferences of all engineers and then send everyone who wants to be notified at that point a notification, via methods of their own choosing (SMS, App, Phone call). Engineers can then *acknowledge* the alert if they are able to provide assistance, *or do nothing* if they can't. After an hour (or a configurable time period later!), if nobody has acknowledged the alert, it'll automatically escalate to a (opt-in) group of second tier folks who can respond.

We're going to play with pagerduty rules to try make this possible.

However, even without *any* of this, it still provides value by streamlining the 'incident response' process and removing a medium (GitHub) that currently is a bit of a bottleneck - and instead moving it all to slack.

**Comment by yuvipanda** [comment]
Removing myself as I'm not currently working on it. #1804 is related though, and I *am* working on that.

**Comment by yuvipanda** [comment]
However, to prevent the perfect from being the enemy of the good, please do consider that *anyone* on the team can always reach out to me at any time to escalate an outage.

**Comment by damianavila** [comment]
Related: https://github.com/2i2c-org/team-compass/issues/763
